frontend/test/views/AccountView.spec.js

```diff
@@ -83,34 +88,85 @@ describe('AccountView.vue', () => {
   it('should toggle the secret value on click', async () => {
     const wrapper = getMountedComponent();
 
-    let sectionBox = wrapper.findComponent('[data-test=credentials]');
+    let credentialsBox = wrapper.findComponent('[data-test=credentials]');
 
-    await sectionBox.trigger('click');
+    await credentialsBox.vm.$emit('display-secret');
     await wrapper.vm.$nextTick();
 
-    sectionBox = wrapper.findComponent('[data-test=credentials]');
-    expect(sectionBox.props('content')[1].value).toEqual([wrapper.vm.user.clientSecret]);
+    credentialsBox = wrapper.findComponent('[data-test=credentials]');
+    expect(credentialsBox.props('clientSecret')).to.equal(wrapper.vm.clientSecret);
   });
 
   it('should not toggle the secret value on click again after it has been toggled', async () => {
     const wrapper = getMountedComponent();
 
-    let sectionBox = wrapper.findComponent('[data-test=credentials]');
+    let credentialsBox = wrapper.findComponent('[data-test=credentials]');
+
+    await credentialsBox.vm.$emit('display-secret');
+    await wrapper.vm.$nextTick();
+
+    credentialsBox = wrapper.findComponent('[data-test=credentials]');
+    expect(credentialsBox.props('clientSecret')).to.equal(wrapper.vm.clientSecret);
+    await credentialsBox.vm.$emit('display-secret');
+    await wrapper.vm.$nextTick();
+
+    credentialsBox = wrapper.findComponent('[data-test=credentials]');
+    expect(credentialsBox.props('clientSecret')).to.equal(wrapper.vm.clientSecret);
+  });
+
+  it('should call onGenerateSecret method and update clientSecret', async () => {
+    getAPICredentialsStub.returns({client_secret: 'newSecret'});
+
+    const wrapper = getMountedComponent();
+    const credentialsBox = wrapper.findComponent('[data-test=credentials]');
+
+    await credentialsBox.vm.$emit('generateSecret');
+    await wrapper.vm.$nextTick();
+
+    expect(generateSecretStub.called).to.be.true;
+    expect(getAPICredentialsStub.called).to.be.true;
+    expect(wrapper.vm.clientSecret).to.equal('newSecret');
```

Comment:
Need to check the prop change that **clientSecret** is used within the computed **secretValue** instead of the data attribute directly.

Reply:
I observed this fix. Am resolving but noticed this elsewhere; have left a comment.